### PR TITLE
Move contrib.render functionality to infer.inspect

### PR DESCRIPTION
**Pull request by mattja**
Moving the `get_model_relations()` and `render_model()` to `numpyro.infer.inspect`, matching pyro interface.
Also updating documentation and tests accordingly.

### CODE REVIEW

File: numpyro/contrib/render.py [hunk 1]
```diff
@@ -1,385 +0,0 @@
-# Copyright Contributors to the Pyro project.
-# SPDX-License-Identifier: Apache-2.0
-
-import itertools
```
Comment (fehiepsi) [inline numpyro/contrib/render.py:4]: Can we keep this file with a single line for backward-compatibility.
```
from numpyro.infer.inspect import render
```

File: numpyro/__init__.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
 import logging
 
 from numpyro import compat, diagnostics, distributions, handlers, infer, ops, optim
-from numpyro.contrib.render import render_model
+from numpyro.infer.inspect import render_model
```
Comment (fehiepsi) [inline numpyro/__init__.py:7]: nit: run `isort` and swap line 7 and 8

File: docs/source/utilities.rst [hunk 3]
```diff
@@ -101,15 +101,9 @@ Model inspection
...
     :member-order: bysource
 
-.. autofunction:: numpyro.contrib.render.get_model_relations
-
 Visualization Utilities
 =======================
 
-render_model
-------------
-.. autofunction:: numpyro.contrib.render.render_model
```
Comment (fehiepsi) [inline docs/source/utilities.rst:111]: Can we keep this entry with the new path?

**Review by fehiepsi — COMMENTED** [review]
(sorry for the late response!)

Thanks for the refactoring to keep API similar to Pyro! I just have a few minor comments

File: numpyro/contrib/render.py (hunk 1 above)
Comment (mattja) [reply numpyro/contrib/render.py:4]: Done. importing the four functions here for full backwards compatibility.

File: numpyro/__init__.py (hunk 2 above)
Comment (mattja) [reply numpyro/__init__.py]: Fixed this and the other 2 lint errors.

File: docs/source/utilities.rst (hunk 3 above)
Comment (mattja) [reply docs/source/utilities.rst:111]: Done. Turned off automodule for `infer.inspect` to avoid duplicate entry.

**Review by fehiepsi — APPROVED** [review]
Thanks, @mattja!

**Pull request merged**